mava/wrappers/env_preprocess_wrappers.py

```diff
@@ -213,3 +218,137 @@ def __getattr__(self, name: str) -> Any:
             return self.__getattribute__(name)
         else:
             return getattr(self._environment, name)
+
+
+class StackObservations:
+    """Stack frames of observations together"""
+
+    def __init__(self, environment: Any, num_frames: int = 1):
+        """Intialise the environment and observation frames"""
+
+        self._environment = environment
+        self.frames: Any = {
+            agent: deque([], maxlen=num_frames)
+            for agent in self._environment.possible_agents
+        }
+        self.num_frames = num_frames
+
+    def reset(self) -> dm_env.TimeStep:
+        """Reset the environment by setting the same observation to all the frames"""
+
+        timestep = self._environment.reset()
+        if type(timestep) == tuple:
```

Comment:
I think we should spend some time trying to design simple, composable wrappers similar to how supersuit does this. They have a general wrapper class and each subclass modifies it wants , e.g. [here](https://github.com/Farama-Foundation/SuperSuit/blob/314831a7d18e7254f455b181694c049908f95155/supersuit/generic_wrappers/frame_stack.py#L50) where frame stracking only changes the `modify_obs` function. That way different wrappers can extend this base class.

What do you think @ulricharmel ?

Reply:
@KaleabTessera that's good idea. I will check what they do in depth.